### ISSUE TITLE
Switch to categorical handling when color_key specified

### ISSUE THREAD

**Issue by jlstevens** [holoviz/hvplot#938]
This issue is one of a number that relate to datashader support of timeseries.

```
>>> hv.__version__, bokeh.__version__, ds.__version__, panel.__version__, hvplot.__version__
('1.15.1', '2.4.3', '0.14.2', '0.13.1', '0.8.1')
```

```python
>>> timestamps = [pd.Timestamp('2022-04-01 0{0}:00:00-0000'.format(hour)) for hour in range(10)] 
>>> df = pd.DataFrame({'timestamp':timestamps, 'SPY':np.random.rand(10), 'NDAQ':np.random.rand(10)})
>>> melted = pd.melt(df, id_vars=['timestamp'], var_name='stock', value_name='high')
>>> melted['bond_indicator'] = melted['stock'].isin(['SPY'])
```

There are multiple problems illustrated here due to `color_key` not forcing categorical handling:

<img width="1096" alt="image" src="https://user-images.githubusercontent.com/890576/195603748-2b13398d-4163-40e6-a65b-eff618716d58.png">

1. There is a colorbar when there shouldn't be one as this should be categorical (only two colors, red and green)
2. There is a legend but it isn't actually working properly (it is blank). This may just be bokeh having difficult showing legends for images.
3.  There is a mixture of red and green for all the lines! This is because instead of being treated as a categorical, the curves are being antialiased (`line_width=1`) which gives you floats, with a stronger value in the middle of the line and lower values near the antialiased, fuzzy edges.

By using `line_width=0` it is possible to see that things weren't actually called correctly (only red lines):

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/890576/195604854-1a4bb1cb-3855-4d2a-9f59-59ff5682edae.png">

This issue is WIP as I'll be filing some related issues shortly and making some edits...

**Comment by jbednar** [comment]
I agree that both `color` and `color_key` should indicate categorical unless otherwise overridden. I'm not sure what should force non-categorical processing.
